### PR TITLE
CryptoDeleteTransitionLogic refactoring

### PR DESCRIPTION
**Pull request by CordonaCodeCraft**
## Purpose
This PR refactors the `CryptoDeleteTransitionLogic` and aligns the implementation with the negotiated new approach of using the new models.

## Notes to the reviewer 
A small, but significant refactoring is the new implementation of the private `validateUsable()` method at `AccuntStore.class`, which is being called by the `AccountStore.loadAccount()` method. 

The current implementation:

image

is being replaced by this new approach:

image

The newly introduced `boolean`  `accountIsDetached` is a direct replica of `HederaLedger`'s `isDetached()` public method and therefore the `validateUsable()` exhaustively validates, that the account is usable without the need for additional and partial validations like 

image

Due to the fact, that the `AccountStore.loadAccount()` is being called by many transition logics - all the E2E tests in the crypto series have been run to validate, that the new implementation satisfies the code base.  

**Addresses**: #367

- [x] E2E passing
- [x] Unit tests passing

### CODE REVIEW

File: hedera-node/src/main/java/com/hedera/services/store/TypedTokenStore.java [hunk 1]
```diff
@@ -149,13 +156,10 @@ public long currentMintedNfts() {
 	 * in order for its changes to be applied to the Swirlds state, and included in the
 	 * {@link com.hedera.services.state.submerkle.ExpirableTxnRecord} for the active transaction.
 	 *
-	 * @param token
-	 * 		the token in the relationship to load
-	 * @param account
-	 * 		the account in the relationship to load
```
Comment (Daniel-K-Ivanov) [inline hedera-node/src/main/java/com/hedera/services/store/TypedTokenStore.java:155]: nit: I think that those changes for the javadoc are auto formatted by a tool. Lets revert those changes and ask in the weekly sync the tool that they use for the java doc

File: hedera-node/src/main/java/com/hedera/services/txns/token/TokenLogicModule.java [hunk 2]
```diff
@@ -52,118 +52,124 @@
 
 @Module
 public abstract class TokenLogicModule {
```
Comment (Daniel-K-Ivanov) [inline hedera-node/src/main/java/com/hedera/services/txns/token/TokenLogicModule.java:54]: nit: lets revert the changes to the formatting

File: hedera-node/src/test/java/com/hedera/services/store/TypedTokenStoreTest.java [hunk 3]
```diff
@@ -109,8 +106,9 @@ void setUp() {
 				() -> tokenRels,
 				backingTokenRels,
 				uniqTokenViewsManager,
-				tokenStore::addKnownTreasury,
-				legacyStore::removeKnownTreasuryForToken);
+				legacyStore::addKnownTreasury,
+				legacyStore::removeKnownTreasuryForToken,
+				legacyStore::isKnownTreasury);
```
Comment (Daniel-K-Ivanov) [inline hedera-node/src/test/java/com/hedera/services/store/TypedTokenStoreTest.java:111]: instead of using 3 different "functional interfaces", lets define 1 interface with 3 methods?

Comment (joan41868) [reply hedera-node/src/test/java/com/hedera/services/store/TypedTokenStoreTest.java:111]: I don't think this is possible, as functional interfaces are interfaces with **single abstract method** , which can be implemented by a lambda function, as here.

**Review by georgiyazovaliiski — APPROVED** [review]

**Review by alshapi — APPROVED** [review]

File: hedera-node/src/main/java/com/hedera/services/store/TypedTokenStore.java [hunk 4]
```diff
@@ -515,4 +507,9 @@ private void alertTokenBackingStoreOfNew(TokenRelationship newRel) {
 	public interface LegacyTreasuryRemover {
 		void removeKnownTreasuryForToken(final AccountID aId, final TokenID tId);
 	}
+
+	@FunctionalInterface
```
Comment (joan41868) [inline hedera-node/src/main/java/com/hedera/services/store/TypedTokenStore.java:511]: **NIT**

You can check and try implementing a similar approach:
https://github.com/LimeChain/hedera-services/blob/60610a6212b97cd9b49cdd5a67df9dc7d83168da/hedera-node/src/main/java/com/hedera/services/store/schedule/ExtantSchedulesDelegator.java

**Review by joan41868 — COMMENTED** [review]
Overall - looks great. I've left a suggestion in the comments :)

**Review by Daniel-K-Ivanov — APPROVED** [review]
LGTM!
